CONTRIBUTING.md

```diff
@@ -4,6 +4,47 @@
 1. All python code must be python 2.6/2.7/3.6 compatible
 1. The code should follow linting from pylint
 
+## Prerequisites
+
+### Install pre-commit
+
+We use `pre-commit` to run code formatting/linting tools on each commit. `pre-commit` operates only on **staged** files, and will only run checks that are applicable to files that have changes staged.
+
+```console
+# Ubuntu/Fedora
+# (installs pre-commit to $HOME/bin - make sure this is in your PATH)
+$ curl https://pre-commit.com/install-local.py | python -
+
+# macOS
+$ brew install pre-commit
+```
+
+If you want to format and verify your code passes on `git commit`, you can install pre-commit hooks using:
```

Comment:
This makes it sound like it's not necessary to do. If you want pre-commit to work you have to install it. We're still talking about `pre-commit` at this point in the file so should be changed to an instruction on how to enable it